### DISCUSSION TITLE
SSL error when using HTTPS

### DISCUSSION THREAD

**Discussion by JKnowles26** [mitmproxy/mitmproxy#5867] [Q&A]
Hey! I'm sure this is something simple I'm missing, but I've been looking into it for a while now and I can't figure it out.

I'm having an issue where when I request an URL using an HTTPs request, I get the following error message on the proxy-side:

```
[17:36:58.764] Addon error: Traceback (most recent call last):
  File "/home/jacob_knowles/.local/lib/python3.9/site-packages/mitmproxy/addons/tlsconfig.py", line 178, in tls_start_client
    tls_start.ssl_conn.use_certificate(entry.cert.to_pyopenssl())
  File "/home/jacob_knowles/.local/lib/python3.9/site-packages/OpenSSL/SSL.py", line 1658, in __getattr__
    raise AttributeError(
AttributeError: 'Connection' object has no attribute 'use_certificate
```

And on the client side:

```
me@my_computer test % curl --proxy 34.133.109.231:10001 "https://www.google.com"
curl: (35) LibreSSL SSL_connect: SSL_ERROR_SYSCALL in connection to www.google.com:443 
```

Or this if I'm trying to send a GET request through code:

```
aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host www.google.com:443 ssl:False [None]
```

It's also worth mentioning that I've written an addon that loads an instance of selenium and passes the request to that to use with an another proxy address. If anymore information might be useful please let me know!

**Comment by mhils** [comment]
This error indicates that you are using an outdated version of cryptography and/or pyOpenSSL.
